e2e/app/aou-elements/xpath-finder.ts

```diff
@@ -0,0 +1,114 @@
+import {ElementHandle, Page, WaitForSelectorOptions} from 'puppeteer';
+import TextOptions from './text-options';
+import * as xpathDefaults from './xpath-defaults';
+
+/**
+ * Find a LINK or BUTTON element with a specified label.
+ * @param {string} label
+ */
+export async function findClickable(page: Page, label: string, options?: WaitForSelectorOptions): Promise<ElementHandle> {
+  const selector = xpathDefaults.clickableXpath(label);
+  return page.waitForXPath(selector, options);
+}
+
+/**
+ * Find SELECT element with a specified label.
+ * @param {string} label
+ */
+export async function findSelect(page: Page, textOptions: TextOptions, waitOptions?: WaitForSelectorOptions): Promise<ElementHandle> {
+  if (textOptions.ancestorNodeLevel === undefined) {
+    textOptions.ancestorNodeLevel = 2;
```

Comment:
Per prior comment - this seems arbitrary, please add a comment explaining why this would default to 2

Reply:
👍 I added comment. `2` is chosen because it is the majority of cases.
```
// ancestor node level is used to find the closest common parent for the label element and input element.
// For most cases, closest parent element is two level up from label. Thus for the default value 2.
```